### ISSUE TITLE
Empty Magazines Unable to be Used in Blueprints

### ISSUE THREAD

**Issue by lucas4114** [SmartlyDressedGames/Unturned-3.x-Community#4448]
Magazine items are unable to be used in blueprints if their (ammo) amount is 0/1, despite the item(s) being present in the inventory.

What I did to produce this issue:

I changed Swissgewehr_20.dat to:
```
GUID 41bd2b68446e497caed8954aba3b8f9c
Type Magazine
Rarity Rare
ID 1490

Size_X 1
Size_Y 2
Size_Z 0.3

Amount 20
Count_Min 5
Count_Max 15

Calibers 1
Caliber_0 1

Tracer 48

Blueprints 5
Blueprint_0_Type Ammo
Blueprint_0_Supplies 1
Blueprint_0_Supply_0_ID 43
Blueprint_0_Build 30
Blueprint_1_Type Ammo
Blueprint_1_Supplies 1
Blueprint_1_Supply_0_ID 6
Blueprint_1_Build 30
Blueprint_2_Type Ammo
Blueprint_2_Supplies 1
Blueprint_2_Supply_0_ID 17
Blueprint_2_Build 30
Blueprint_3_Type Ammo
Blueprint_3_Supplies 1
Blueprint_3_Supply_0_ID 1490
Blueprint_3_Build 30

//Scrap
Blueprint_4_Type Gear
Blueprint_4_Supplies 1
Blueprint_4_Supply_0_ID 1490
Blueprint_4_Product 67
Blueprint_4_Products 1
Blueprint_4_Level 2
Blueprint_4_Skill Craft
Blueprint_4_Build 30

Paintable
```

Screenshots demonstrating the issue: 

I maxed out my "Crafting" skill (just showing this step to confirm that I did it):

![Img12](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/21cf72d8-4ea8-44dd-a55b-1fb3946c146c)

With empty magazines present in the inventory, the blueprint is unavailable:

![Img1](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/12d75544-faf2-443e-845a-75643e6d59eb)

![Img2](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/18fe2dd8-e119-451f-badc-69c63be99ad3)

![Img3](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/c4ca272a-d6d2-45b0-98e9-2e34dfcde665)

If a magazine is full, then the blueprint works correctly for that item only:

![Img4](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/bef2ce3b-c749-4df3-b225-ae22797cc619)

![Img5](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/5c284483-b420-477f-8076-6441ef25a108)

If a magazine is partially full (ammo amount is > 0), then it also works normally for that item only:

![Img7](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/e130deea-449b-46af-843f-e83dd34e795f)

![Img6](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/0f1a974d-c395-46e8-908f-a82d1ec4f97a)

...

Some things I tried to fix this:

1. I tried changing `Blueprint_#_Type Gear` to `Blueprint_#_Type Ammo`

```
GUID 41bd2b68446e497caed8954aba3b8f9c
Type Magazine
Rarity Rare
ID 1490

Size_X 1
Size_Y 2
Size_Z 0.3

Amount 20
Count_Min 5
Count_Max 15

Calibers 1
Caliber_0 1

Tracer 48

Blueprints 5
Blueprint_0_Type Ammo
Blueprint_0_Supplies 1
Blueprint_0_Supply_0_ID 43
Blueprint_0_Build 30
Blueprint_1_Type Ammo
Blueprint_1_Supplies 1
Blueprint_1_Supply_0_ID 6
Blueprint_1_Build 30
Blueprint_2_Type Ammo
Blueprint_2_Supplies 1
Blueprint_2_Supply_0_ID 17
Blueprint_2_Build 30
Blueprint_3_Type Ammo
Blueprint_3_Supplies 1
Blueprint_3_Supply_0_ID 1490
Blueprint_3_Build 30

//Scrap
Blueprint_4_Type Ammo
Blueprint_4_Supplies 1
Blueprint_4_Supply_0_ID 1490
Blueprint_4_Product 67
Blueprint_4_Products 1
Blueprint_4_Level 2
Blueprint_4_Skill Craft
Blueprint_4_Build 30

Paintable
```

Result:

With only empty magazines in the inventory, the blueprint was unavailable.

![Img8](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/7d1114ad-6cb3-4f2c-98eb-4aecb681a4ea)

With a magazine with > 0 ammo present in the inventory, the blueprint was available but didn't do anything when clicked (broken).

![Img9](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/90570499-ed03-4387-9aa1-63b64ca92f8b)

![Img10](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/c7cfd91d-9eb2-47e7-bef5-ab15b565e816)

2. I tried changing `Blueprint_#_Type Gear` to `Blueprint_#_Type Tool`

```
GUID 41bd2b68446e497caed8954aba3b8f9c
Type Magazine
Rarity Rare
ID 1490

Size_X 1
Size_Y 2
Size_Z 0.3

Amount 20
Count_Min 5
Count_Max 15

Calibers 1
Caliber_0 1

Tracer 48

Blueprints 5
Blueprint_0_Type Ammo
Blueprint_0_Supplies 1
Blueprint_0_Supply_0_ID 43
Blueprint_0_Build 30
Blueprint_1_Type Ammo
Blueprint_1_Supplies 1
Blueprint_1_Supply_0_ID 6
Blueprint_1_Build 30
Blueprint_2_Type Ammo
Blueprint_2_Supplies 1
Blueprint_2_Supply_0_ID 17
Blueprint_2_Build 30
Blueprint_3_Type Ammo
Blueprint_3_Supplies 1
Blueprint_3_Supply_0_ID 1490
Blueprint_3_Build 30

//Scrap
Blueprint_4_Type Tool
Blueprint_4_Supplies 1
Blueprint_4_Supply_0_ID 1490
Blueprint_4_Product 67
Blueprint_4_Products 1
Blueprint_4_Level 2
Blueprint_4_Skill Craft
Blueprint_4_Build 30

Paintable
```

Result:

Changing `Blueprint_#_Type Gear` to `Blueprint_#_Type Tool` has the same behaviour as as `Blueprint_#_Type Gear`, except the blueprint appears in the "Tools" crafting tab.

Magazines with 0 ammo amount can't be used in the blueprint. Magazines with > 0 ammo amount can be used in the blueprint.

3. I tried removing `Blueprint_#_Supplies` and `Blueprint_#_Supply_#_ID` lines in hopes that the blueprint system would automatically use its own item as the supply.

```
GUID 41bd2b68446e497caed8954aba3b8f9c
Type Magazine
Rarity Rare
ID 1490

Size_X 1
Size_Y 2
Size_Z 0.3

Amount 20
Count_Min 5
Count_Max 15

Calibers 1
Caliber_0 1

Tracer 48

Blueprints 5
Blueprint_0_Type Ammo
Blueprint_0_Supplies 1
Blueprint_0_Supply_0_ID 43
Blueprint_0_Build 30
Blueprint_1_Type Ammo
Blueprint_1_Supplies 1
Blueprint_1_Supply_0_ID 6
Blueprint_1_Build 30
Blueprint_2_Type Ammo
Blueprint_2_Supplies 1
Blueprint_2_Supply_0_ID 17
Blueprint_2_Build 30
Blueprint_3_Type Ammo
Blueprint_3_Supplies 1
Blueprint_3_Supply_0_ID 1490
Blueprint_3_Build 30

//Scrap
Blueprint_4_Type Gear
Blueprint_4_Product 67
Blueprint_4_Products 1
Blueprint_4_Level 2
Blueprint_4_Skill Craft
Blueprint_4_Build 30

Paintable
```

Result: the blueprint was not in the UI.

![Img11](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/57552571/c1075928-ba5c-4dfe-9aae-0088b24bde30)

4. I removed `Blueprint_#_Supplies` and `Blueprint_#_Supply_#_ID` lines again and changed the `Blueprint_#_Type Gear` line to `Blueprint_#_Type Tool`.

```
GUID 41bd2b68446e497caed8954aba3b8f9c
Type Magazine
Rarity Rare
ID 1490

Size_X 1
Size_Y 2
Size_Z 0.3

Amount 20
Count_Min 5
Count_Max 15

Calibers 1
Caliber_0 1

Tracer 48

Blueprints 5
Blueprint_0_Type Ammo
Blueprint_0_Supplies 1
Blueprint_0_Supply_0_ID 43
Blueprint_0_Build 30
Blueprint_1_Type Ammo
Blueprint_1_Supplies 1
Blueprint_1_Supply_0_ID 6
Blueprint_1_Build 30
Blueprint_2_Type Ammo
Blueprint_2_Supplies 1
Blueprint_2_Supply_0_ID 17
Blueprint_2_Build 30
Blueprint_3_Type Ammo
Blueprint_3_Supplies 1
Blueprint_3_Supply_0_ID 1490
Blueprint_3_Build 30

//Scrap
Blueprint_4_Type Tool
Blueprint_4_Product 67
Blueprint_4_Products 1
Blueprint_4_Level 2
Blueprint_4_Skill Craft
Blueprint_4_Build 30

Paintable
```

Result: the blueprint was not in the UI.

_Basically,_ there is no workaround that I could think of yet...

Conclusion:

Empty magazines cannot be used in blueprints at all.

If empty magazines were able to be used in blueprints, then various mods become possible, such as: using magazines for crafting other (new) magazines (like a "double" magazine for example), using magazines to craft other items, and salvaging magazines into scrap metal (which was my intention).

**Comment by SDGNelson** [comment]
Sorry to say this is unlikely to change until after a crafting rewrite. Regular item crafting adds up the item "amount" to determine how many of an item you have, whereas ammo containers use amount to represent the number of contained bullets, so regular item crafting thinks you have zero magazines.
